server.go

```diff
@@ -311,76 +312,28 @@ func (s *Server) monitorMaxSlices() {
 
 // ReceiveMessage represents an implementation of BroadcastHandler.
 func (s *Server) ReceiveMessage(pb proto.Message) error {
-	switch obj := pb.(type) {
-	case *internal.CreateSliceMessage:
-		idx := s.Holder.Index(obj.Index)
-		if idx == nil {
-			return fmt.Errorf("Local Index not found: %s", obj.Index)
-		}
-		if obj.IsInverse {
-			idx.SetRemoteMaxInverseSlice(obj.Slice)
-		} else {
-			idx.SetRemoteMaxSlice(obj.Slice)
-		}
-	case *internal.CreateIndexMessage:
-		opt := IndexOptions{
-			ColumnLabel: obj.Meta.ColumnLabel,
-			TimeQuantum: TimeQuantum(obj.Meta.TimeQuantum),
-		}
-		_, err := s.Holder.CreateIndex(obj.Index, opt)
-		if err != nil {
-			return err
-		}
-	case *internal.DeleteIndexMessage:
-		if err := s.Holder.DeleteIndex(obj.Index); err != nil {
-			return err
-		}
-	case *internal.CreateFrameMessage:
-		idx := s.Holder.Index(obj.Index)
-		if idx == nil {
-			return fmt.Errorf("Local Index not found: %s", obj.Index)
-		}
-		opt := FrameOptions{
-			RowLabel:       obj.Meta.RowLabel,
-			InverseEnabled: obj.Meta.InverseEnabled,
-			RangeEnabled:   obj.Meta.RangeEnabled,
-			CacheType:      obj.Meta.CacheType,
-			CacheSize:      obj.Meta.CacheSize,
-			TimeQuantum:    TimeQuantum(obj.Meta.TimeQuantum),
-			Fields:         decodeFields(obj.Meta.Fields),
-		}
-		_, err := idx.CreateFrame(obj.Frame, opt)
-		if err != nil {
-			return err
-		}
-	case *internal.DeleteFrameMessage:
-		idx := s.Holder.Index(obj.Index)
-		if err := idx.DeleteFrame(obj.Frame); err != nil {
-			return err
-		}
-	case *internal.CreateInputDefinitionMessage:
-		idx := s.Holder.Index(obj.Index)
-		if idx == nil {
-			return fmt.Errorf("Local Index not found: %s", obj.Index)
-		}
-		idx.CreateInputDefinition(obj.Definition)
-	case *internal.DeleteInputDefinitionMessage:
-		idx := s.Holder.Index(obj.Index)
-		err := idx.DeleteInputDefinition(obj.Name)
+	return s.Handler.ProcessClusterMessage(pb)
+}
+
+// SendSync represents an implementation of BroadcastHandler.
+func (s *Server) SendSync(pb proto.Message) error {
+	var eg errgroup.Group
+	for _, node := range s.Cluster.Nodes {
```

Comment:
Just a note here that we need to confirm that limiting `SendSync` messages to nodes registered in the cluster (as opposed to including all nodes that memberlist is aware of) isn't a problem in the resize logic.